httpbp/client_middlewares.go

```diff
@@ -349,3 +355,47 @@ func PrometheusClientMetrics(serverSlug string) ClientMiddleware {
 		})
 	}
 }
+
+type ServiceAddressParts struct {
+	Name      string
+	Namespace string
+}
+
+func FaultInjection() ClientMiddleware {
+	return func(next http.RoundTripper) http.RoundTripper {
+		return roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+			resumeFn := faults.ResumeFn(func() (interface{}, error) {
+				return next.RoundTrip(req)
+			})
+			responseFn := faults.ResponseFn(func(code int, message string) (interface{}, error) {
+				return &http.Response{
+					Status:     http.StatusText(code),
+					StatusCode: code,
+					Proto:      req.Proto,
+					ProtoMajor: req.ProtoMajor,
+					ProtoMinor: req.ProtoMinor,
+					Header: map[string][]string{
+						// Copied from the standard http.Error() function.
+						"Content-Type":           {"text/plain; charset=utf-8"},
+						"X-Content-Type-Options": {"nosniff"},
+					},
+					ContentLength:    0,
+					TransferEncoding: req.TransferEncoding,
+					Request:          req,
+					TLS:              req.TLS,
+				}, nil
+			})
+
+			resp, err := faults.InjectFault(faults.InjectFaultParams{
+				CallerName:   "httpbp.FaultInjection",
+				Address:      req.URL.Host,
+				Method:       strings.TrimPrefix(req.URL.Path, "/"),
+				AbortCodeMin: 100,
```

Comment:
shouldn't min be 400?

Reply:
Originally I was thinking this could be flexible to allow other response codes, but it really doesn't make sense given the actual response returned when a fault is injected. Thanks for raising this. Updated to 400.